internal/pkg/cli/svc_override.go

```diff
@@ -120,6 +154,109 @@ func (o *overrideSvcOpts) validateCDKLang() error {
 		strings.Join(applyAll(validCDKLangs, strconv.Quote), ", "))
 }
 
+func (o *overrideSvcOpts) validateOrAskServiceName() error {
+	if o.name == "" {
+		return o.askServiceName()
+	}
+	return o.validateServiceName()
+}
+
+func (o *overrideSvcOpts) validateServiceName() error {
+	names, err := o.ws.ListServices()
+	if err != nil {
+		return fmt.Errorf("list services in the workspace: %v", err)
+	}
+	if !contains(o.name, names) {
+		return fmt.Errorf("service %q does not exist in the workspace", o.name)
+	}
+	return nil
+}
+
+func (o *overrideSvcOpts) askServiceName() error {
+	name, err := o.wsPrompt.Service("Which service's resources would you like to override?", "")
+	if err != nil {
+		return fmt.Errorf("select service name from workspace: %v", err)
+	}
+	o.name = name
+	return nil
+}
+
+func (o *overrideSvcOpts) validateOrAskIaCTool() error {
+	if o.iacTool == "" {
+		return o.askIaCTool()
+	}
+	return o.validateIaCTool()
+}
+
+func (o *overrideSvcOpts) validateIaCTool() error {
+	for _, valid := range validIaCTools {
+		if o.iacTool == valid {
+			return nil
+		}
+	}
+	return fmt.Errorf("%q is not a valid IaC tool: must be one of: %s",
+		o.iacTool,
+		strings.Join(applyAll(validIaCTools, strconv.Quote), ", "))
+}
+
+func (o *overrideSvcOpts) askResourcesToOverride() error {
+	if len(o.resources) != 0 {
+		return nil
+	}
+
+	buf := &closableStringBuilder{
+		Builder: new(strings.Builder),
+	}
+	pkgCmd, err := o.packageCmd(buf)
+	if err != nil {
+		return err
+	}
+	if err := pkgCmd.Execute(); err != nil {
+		return fmt.Errorf("generate CloudFormation template for service %q: %v", o.name, err)
+	}
+	msg := fmt.Sprintf("Which resources in %q would you like to override?", o.name)
+	resources, err := o.cfnPrompt.Resources(msg, "Resources:", "", buf.String())
+	if err != nil {
+		return err
+	}
+	o.resources = resources
+	return nil
+}
+
+func (o *overrideSvcOpts) newSvcPackageCmd(tplBuf stringWriteCloser) (executor, error) {
+	envs, err := o.cfgStore.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("list environments in application %q: %v", o.appName, err)
+	}
+	if len(envs) == 0 {
+		return nil, fmt.Errorf("no environments found in application %q", o.appName)
+	}
+	cmd, err := newPackageSvcOpts(packageSvcVars{
+		name:    o.name,
+		envName: envs[0].Name, // Use any environment to get a sample CloudFormation template for the service.
```

Comment:
oh yea great question!

`svc override` is "best effort" so the idea is we'll just scaffold some resources for you to make it easy to get started but you could skip this command altogether and write the files yourself manually.

For example, to be able to do the example you wrote above, in the CDK app's `stack.ts` the client would write:

```ts
if props.envName === "prod" {
   this.transformEFS();
}
```

Reply:
+1 to Austin's question. Maybe we should take env as an input?

Reply:
Yeah, I think I like adding env as an input here. That lets the customer get the correct, expected template for the given inputs and there's no guesswork. If there's only one environment, then there will be no selector.

Reply:
But isn't the end outcome the same? For example, as a user I choose "prod" and the resource is available there but then missing in the "test" environment. That doesn't really help me, I have to write the if-statement either way.

I am down to add the `--env` flag as optional, but I'm still not in favor of prompting here.

Reply:
Isn't the final manifest itself determined by the environment after applying the env specific setting? If we don't know what environment they are trying to override the service against, how do we generate the correct CFN template and get all resources 🤔

Reply:
hmm, I think a lot depends on the mechanics of how overrides are implemented. I wonder if there is a way to help customers understand the differences in templates between their environments so they can write correct CDK code. Visibility into this would be useful. Maybe providing a oneliner to "svc package > a.yml && svc package > b.yml && diff a.yml b.yml". Or a guide. Otherwise it's a little like a shot in the dark which resources are present in which envs.

Reply:
Fact is this is a high context command which requires a lot of understanding, so perhaps we can provide plenty of education and insight into how someone might use it. That's for the site and the blog, though, and shouldn't block us here.

Reply:
If the customer is indeed doing CloudFormation's best practice: 
> Reuse templates to replicate stacks in multiple environments
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/best-practices.html#reuse

Then the environment does not matter. This is why I'm not prompting for the environment.

If the user is using `environments` in the manifest to toggle resources, then even if we were to prompt for the environment to show the "right" resources the `package` command will fail while deploying in other environments.
So is it really important to show the appropriate environment resources, I'm still not convinced?

I'd rather optimize for the case where `copilot package` would always output the same resources across environments.

Reply:
🧐 I see, so this edge case is _possible_ but not _likely_, nor is it best practice. I wonder if there's a natural place in the docs to advise against toggling resources with overrides, or if this is a niche enough problem to just disregard it. 

I'm convinced by this; otherwise the pr looks good! 

:shipit:

Reply:
I'll add the flag as optional, so that folks have a way of seeing the resource that they want :) but I just don't think that different templates/env is a common enough scenario to warrant making `--env` _required_.

Reply:
Created the PR here to add the optional flag: https://github.com/aws/copilot-cli/pull/4573 !